### PR TITLE
Head body length change

### LINKED ISSUE #189: Body length of a response to a HEAD request
In http/af, `Response.body_length` takes a `request_method` argument that should be the method of its corresponding request. One reason for this is to set the body length of HEAD messages to be `` `Fixed 0L ``. I think it would be a good idea to also do this in h2.

From RFC7540§8.1.2.6:

> A response that is defined to have no payload, as described in RFC7230, Section 3.3.2, can have a non-zero content-length header field, even though no content is included in DATA frames.

From RFC7230§3.2.2:

> A server MAY send a Content-Length header field in a response to a HEAD request (Section 4.3.2 of RFC7231); a server MUST NOT send Content-Length in such a response unless its field-value equals the decimal number of octets that would have been sent in the payload body of a response if the same request had used the GET method.

From RFC7231§4.3.2

> The HEAD method is identical to GET except that the server MUST NOT send a message body in the response (i.e., the response terminates at the end of the header section).

As the content-length header for responses to HEAD requests is not the length of body, I think it makes sense to return `` `Fixed 0L ``. If you agree, I can make a pull request to add this behaviour.

### PR DESCRIPTION
**Pull request by jonathanjameswatson**
Resolves #189

### CODE REVIEW

File: lib/h2.mli [hunk 1]
```diff
@@ -454,9 +454,11 @@ module Response : sig
       RFC7540§8.1.2.4} for more details. *)
 
   val body_length
-    :  t
+    :  request_method:Httpaf.Method.standard
```
Comment (anmonteiro) [inline lib/h2.mli:457]:
```suggestion
    :  request_method:Method.standard
```

File: lib_test/test_h2.ml [hunk 2]
```diff
@@ -95,4 +95,62 @@ module Headers_tests = struct
     ]
 end
 
-let () = Alcotest.run "ocaml-h2 unit tests" [ "headers", Headers_tests.suite ]
+module Body_length_tests = struct
```
Comment (anmonteiro) [inline lib_test/test_h2.ml:98]: I didn't see an assertion that a call to ``Response.body_length ~request_method:`HEAD`` returns `` `Fixed 0``

Comment (jonathanjameswatson) [reply lib_test/test_h2.ml:98]: I don't know how I managed to do that

**Comment by anmonteiro** [comment]
Thank you!

**Pull request merged**